website/blog/2023-01-04-ferretdb-beta-version-authentication.md

```diff
@@ -0,0 +1,117 @@
+This new and exciting release is no ordinary milestone; it's a culmination of FerretDB's journey as we work on bringing you the ultimate open-source alternative to MongoDB by converting MongoDB protocol queries to SQL, with PostgreSQL as the database engine.
+
+While we do not aim to cover all the features of MongoDB, our goal with the Beta - and our upcoming GA release later this quarter - is to provide a solid foundation on which to build targeted features that'll enable FerretDB to support more and more real-world use cases.
+Be sure to [check our roadmap for further details](https://github.com/orgs/FerretDB/projects/2).
+
+:::caution
+Please note that this particular change breaks backward compatibility.
+To dump and restore the data, you’ll need connection strings to connect to FerretDB and PostgreSQL.
+There are numerous ways to dump and restore your data.
+For example, you can follow the following steps:
+
+1. Backup FerretDB databases with `mongodump`.
+Set your FerretDB connection string in `—uri` and run:
+   `mongodump --uri="mongodb://localhost:27017"`
+   This command will create a directory with dumps for each FerretDB database on the given server.
+Later, we will use this dump to restore the database.
+Instead of `mongodump`, you can also use `mongoexport` for this.
+
+2. Backup PostgreSQL `ferretdb` database.
+If the migration goes well, we will not use this backup.
+But we recommend to do this step in case you need to rollback.
+Specify your host and port.
+`pg_dump -h localhost -p 5432 -U username ferretdb > ferretdb.sql`
+
+3. Stop FerretDB (This depends on your operating system and the way you run FerretDB)
+
+4. Connect to PostgreSQL and drop `ferredb` database as it's not needed anymore:
+   * `psql -h localhost -p 5432 -U username postgres`
+   * `DROP DATABASE ferretdb`
+
+5. Upgrade FerretDB and run FerretDB 0.8 (Please refer to our documentation where we describe how to update and start FerretDB)
+
+6. Restore database using `mongorestore --uri="mongodb://localhost:27017"`
+
+7. In case something doesn't work and you need to rollback to FerretDB 0.7.1:
+   * Stop ferretdb
+   * Delete ferretdb database (repeat step 4)
+   * Restore `ferretdb` PostgreSQL db from the dump we created on the step 2:  `psql -h localhost -p 5432 -U username ferretdb -f ferretdb.sql`
+   * Start ferretdb 0.7.1
+:::
+
+In this blog post, we'll be sharing detailed information about the FerretDB beta release (0.8.0).
+
+## What's new?
+
+Since the last release (0.7.2), we've made great strides in adding new features to FerretDB, especially the introduction of authentication for PostgreSQL.
+With authentication, users can now connect securely using passwords, ensuring that only authorized connections are established.
+You can implement an authenticated connection by specifying the connection string in this format: `postgres://username:password@postgres:5432/ferretdb`.
```

Comment:
This is how FerretDB connects to postgres, is it what we want to set in the docs?

Reply:
Sorry, I don't understand what you mean here.

Reply:
I mean, why do we mention PostgreSQL connection string in this sentence? What do we expect people do with it?

Reply:
Ok. I was mostly citing the PostgreSQL connection string example for authentication in [our documentation here ](https://docs.ferretdb.io/security/#authentication). Mostly we want to have examples/use cases for each new features in the release. What do you suggest?

Reply:
For example, we can add a whole sentence from the documentation (both PostgreSQL and FerretDB connection strings), as PostgreSQL one without any explanation is hard to interpret.

Reply:
```suggestion
You can implement an authenticated connection to the PostgreSQL database by specifying the default username and password in the FerretDB connection string as `postgres://username:password@postgres:5432/ferretdb`,  or use a MongoDB URI `mongodb://username:password@ferretdb:27018/?tls=true&authMechanism=PLAIN`
```


Is this okay?